esphome/components/opentherm/hub.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+
+#include "esphome/core/defines.h"
+#include "esphome/core/hal.h"
+#include "esphome/core/component.h"
+#include "esphome/core/log.h"
+
+#include "opentherm.h"
+
+#include <unordered_map>
+#include <unordered_set>
+#include <functional>
+
+namespace esphome {
+namespace opentherm {
+
+// OpenTherm component for ESPHome
+class OpenthermHub : public Component {
+ protected:
+  // Communication pins for the OpenTherm interface
+  InternalGPIOPin *in_pin_, *out_pin_;
+  // The OpenTherm interface
+  OpenTherm *opentherm_;
```

Comment:
Since this is only referenced in one place and never shared anywhere else we can safely use a unique ptr here
```suggestion
  std::unique_ptr<OpenTherm> opentherm_;
```

Reply:
Done.